physlr/physlr.py

```diff
@@ -1839,6 +1839,8 @@ def index_minimizers_in_backbones(backbones, bxtomxs):
         mxtopos = {}
         for tid, path in enumerate(progress(backbones)):
             for pos, u in enumerate(path):
+                if u not in bxtomxs:
+                    u = u.rsplit("_", 1)[0]
```

Comment:
After the second round `AAAA_2` would become `AAAA_2_0` and `AAAA_2_1`? It seems a bit clumsy. I'd suggest either `AAAA_2` becomes `AAAA_2` and `AAAA_3` or possibly `AAAA_2` is removed and becomes `AAAA_3` and `AAAA_4`. I have a slight preference for the latter.
If we go with the current naming scheme of `AAAA_2_0`, these two `if` statements should be replaced with a `while` loop `while u not in bxtomxs and "_" in u`

Reply:
Actually I just opened another branch to fix the things downstream so we can fix this. the problem with changing the number is that we need to double check and see how many molecules are there for this one that we are changing its name. For example if there we have `AAAA_0`, `AAAA_1`, and `AAAA_2` then for each of them in the second round we need to check the molecules list. If we deconvolute `AAAA_0` to `AAAA_3` and `AAAA_4` then we need to deconvolute `AAAA_1` to `AAAA_5` and `AAAA_6`. Is that fine? Or alternatively we can stick to the current naming and then go over the output once more to re-number all.

Reply:
and my temporary fix is that we can duplicate the if statement in line 1842 and 1843 so it can go over the current output. I was about to open a PR for that. So let me know what's your suggestion.

Reply:
Oh I did commit that duplicated if statement here... Sorry that was unintentional :D
The problem with while loop is that for stflr we have underscores in the naming, and we already know that we do at most 2 rounds of mol (for now, as it was a temporary fix) so if we change it we a while loop it will do unnecessary operations.

Reply:
You may want to consider changing the underscores in the stLFR barcodes names to say hyphens or some other character to avoid this source of confusion.

Reply:
as it is in the name naming, wouldn't be more logical to change the symbol for molecule separation?
If we change the one in the stflr header we need to process more, if we change underscore in molecule separation then we need no more processing at all. What symbol do you think would be good? hyphen is already being used in the 10x headers. so `#` or `/` ?
and @jowong4 what are your suggestions? as you've already worked with stflr data and you know the symbols they use

Reply:
I think '~' might be fine. I don't recall seeing this symbol used in either of the files and seems like a decent substitute of `-` and `_`

Reply:
anyway, is that fine if with postpone it to another PR and stick with this duplicated if statement?